### PR TITLE
 MongoDB Driver Migration

### PR DESCRIPTION
**Pull request by Marianasilvago**
### What

Update mongo config to add creds and CAfile.
Migrate datastore methods to use new mongo driver library.

### How to review

Check the code changes.

### Who can review

Anyone with understanding of golang and mongo-go-driver library.

### CODE REVIEW

File: mongo/mongo.go [hunk 1]
```diff
@@ -4,46 +4,63 @@ import (
...
 )
 
+const (
+	connectTimeoutInSeconds = 5
+	queryTimeoutInSeconds   = 15
+)
+
 // Mongo represents a simplistic MongoDB configuration.
 type Mongo struct {
 	Session      *mgo.Session
```
Comment (morses-code) [inline mongo/mongo.go:23]: This is leaving references to mgo.

File: mongo/mongo.go [hunk 2]
```diff
@@ -4,46 +4,63 @@ import (
...
-	"github.com/ONSdigital/dp-areas-api/config"
 	"github.com/ONSdigital/dp-areas-api/models"
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
-	dpmongo "github.com/ONSdigital/dp-mongodb"
-	dpMongoHealth "github.com/ONSdigital/dp-mongodb/health"
+	dpMongoHealth "github.com/ONSdigital/dp-mongodb/v2/pkg/health"
+	dpMongoDriver "github.com/ONSdigital/dp-mongodb/v2/pkg/mongo-driver"
 	"github.com/ONSdigital/log.go/log"
 	"github.com/globalsign/mgo"
 	"github.com/globalsign/mgo/bson"
```
Comment (morses-code) [inline mongo/mongo.go:13]: Switch to `go.mongodb.org/mongo-driver/bson`

**Review by morses-code — CHANGES_REQUESTED** [review]
2 small comments, rest LGTM 👍🏻

File: mongo/mongo.go (hunk 2 above)
Comment (Marianasilvago) [reply mongo/mongo.go]: updated it in the new commit.

File: mongo/mongo.go (hunk 1 above)
Comment (Marianasilvago) [reply mongo/mongo.go]: updated it in the new commit.

**Review by morses-code — DISMISSED** [review]
LGTM 👍🏻

**Review by CarlHembrough — DISMISSED** [review]

**Review by CarlHembrough — DISMISSED** [review]

File: mongo/mongo.go [hunk 3]
```diff
@@ -179,9 +188,8 @@ func (m *Mongo) GetAreas(ctx context.Context, offset, limit int) (*models.AreasR
 	return &models.AreasResults{
 		Items:      &values,
 		Count:      len(values),
-		TotalCount: totalCount,
+		TotalCount: int(totalCount),
```
Comment (CarlHembrough) [inline mongo/mongo.go:191]: I don't think the cast to int is required?

**Review by CarlHembrough — CHANGES_REQUESTED** [review]
Changes look good - just the minor point about the cast

**Review by CarlHembrough — APPROVED** [review]

**Pull request merged**